tests/test_utils.py

```diff
@@ -388,66 +399,48 @@ def test_deduplicate_and_sort_reqs(self):
 
     @pytest.fixture
     def temp_test_directory(self) -> Generator:
-        path = os.path.join(os.getcwd(), self.TEMP_TEST_DIRECTORY)
-        os.mkdir(path)
-        yield path
-        rmtree(self.TEMP_TEST_DIRECTORY)  # We delete whole directory in teardown -
+        temp_dir = tempfile.TemporaryDirectory()
+        yield temp_dir.name
+        rmtree(temp_dir.name)  # We delete whole directory in teardown -
         # so, no need to delete any of the files separately
 
     @pytest.fixture
     def temp_directory_for_app_zip(self, temp_test_directory: str) -> Generator:
-        path = os.path.join(temp_test_directory, self.TEMP_DIR_FOR_APP_ZIP)
-        os.mkdir(path)
-        yield path
+        temp_dir = tempfile.TemporaryDirectory(dir=temp_test_directory)
+        yield temp_dir.name
 
     @pytest.fixture
     def correct_app_zip(self, temp_test_directory: str) -> Generator:
-        path = os.path.join(temp_test_directory, self.APP_ZIP)
-        self.create_file(path, [])
-        yield path
+        yield self.create_file(".zip", temp_test_directory, [])
```

Comment:
I wonder if `correct_app_zip` is a good name for function creating an empty file only with `.zip` suffix.